packages/headless/src/features/analytics/analytics-utils.ts

```diff
@@ -265,21 +265,101 @@ function makePreparableAnalyticsAction<
   return rootAction as PreparableAnalyticsAction<EventType, StateNeeded>;
 }
 
-export const makeAnalyticsAction = <
+export type AnalyticsActionOptions<
   EventType extends AnalyticsType,
   StateNeeded extends
     StateNeededBySearchAnalyticsProvider = StateNeededBySearchAnalyticsProvider,
->(
-  prefix: string,
-  analyticsType: EventType,
-  getBuilder: (
+> = LegacyAnalyticsOptions<EventType, StateNeeded> & NextLegacyOptions;
+
+export interface NextLegacyOptions {
+  todo?: never; //Remove when we have at least one property in this interface
+}
+export interface LegacyAnalyticsOptions<
+  EventType extends AnalyticsType,
+  StateNeeded extends
+    StateNeededBySearchAnalyticsProvider = StateNeededBySearchAnalyticsProvider,
+> {
+  prefix: string;
+  analyticsType: EventType;
+  __legacy__getBuilder: (
     client: CoveoSearchPageClient,
     state: StateNeeded
-  ) => Promise<EventBuilder | null> | null,
-  provider: (getState: () => StateNeeded) => SearchPageClientProvider = (
-    getState
-  ) => new SearchAnalyticsProvider(getState)
-): PreparableAnalyticsAction<WrappedAnalyticsType<EventType>, StateNeeded> => {
+  ) => Promise<EventBuilder | null> | null;
+  __legacy__provider?: (
+    getState: () => StateNeeded
+  ) => SearchPageClientProvider;
+}
+
+export function makeAnalyticsAction<
+  EventType extends AnalyticsType,
+  StateNeeded extends
+    StateNeededBySearchAnalyticsProvider = StateNeededBySearchAnalyticsProvider,
+>(
+  prefix: string,
+  analyticsType: LegacyAnalyticsOptions<
+    EventType,
+    StateNeeded
+  >['analyticsType'],
+  __legacy__getBuilder: LegacyAnalyticsOptions<
+    EventType,
+    StateNeeded
+  >['__legacy__getBuilder'],
+  __legacy__provider?: LegacyAnalyticsOptions<
+    EventType,
+    StateNeeded
+  >['__legacy__provider']
+): PreparableAnalyticsAction<WrappedAnalyticsType<EventType>, StateNeeded>;
+export function makeAnalyticsAction<
+  EventType extends AnalyticsType,
+  StateNeeded extends
+    StateNeededBySearchAnalyticsProvider = StateNeededBySearchAnalyticsProvider,
+>({
+  prefix,
+  analyticsType,
+  __legacy__getBuilder,
+  __legacy__provider,
+}: AnalyticsActionOptions<EventType, StateNeeded>): PreparableAnalyticsAction<
+  WrappedAnalyticsType<EventType>,
+  StateNeeded
+>;
+export function makeAnalyticsAction<
+  EventType extends AnalyticsType,
+  StateNeeded extends
+    StateNeededBySearchAnalyticsProvider = StateNeededBySearchAnalyticsProvider,
+>(
+  ...params:
+    | [
+        LegacyAnalyticsOptions<EventType, StateNeeded>['prefix'],
```

Comment:
Going with my above comment about small readability gains, we'd could reuse the temporary types here.